packages/contracts-bedrock/src/libraries/GasPayingToken.sol

```diff
@@ -34,28 +35,33 @@ library GasPayingToken {
         }
     }
 
+    /// @notice Reads the gas paying token's name from the magic storage slot.
+    ///         If nothing is set in storage, then the ether name, 'Ether', is returned instead.
     function getName() internal view returns (string memory name_) {
         (address addr,) = getToken();
         if (addr == Constants.ETHER) {
             name_ = "Ether";
         } else {
-            name_ = string(abi.encodePacked(Storage.getBytes32(GAS_PAYING_TOKEN_NAME_SLOT)));
+            name_ = string(Bytes.trimTrailingZeroes(abi.encodePacked(Storage.getBytes32(GAS_PAYING_TOKEN_NAME_SLOT))));
```

Comment:
Are you sure we need this trim functionality? I am not sure if we need it. What happens without it?

Reply:
we needed it because for example if we store "ETH" as bytes32, when we try to read it from the slot, we obtain a version with a bunch of left padded zeroes. this means that contracts like WETH would return for symbol not "ETH" but "ETH\0\0\0\0\0\0"

Reply:
Can we do this upstream such that we simply need to set/get at this point? ie do the trimming once in the system config initialize